components/SwapPageContent/SwapPageContent.tsx

```diff
@@ -113,46 +114,65 @@ export function SwapPageContent() {
 
   const sqrtPriceAfter = useAsyncValue(async () => {
     if (!amounts) return null;
-    if (amounts.tradeType === TradeType.EXACT_INPUT) {
+    const paprIn = paprTokenField === Field.INPUT;
+    const paprOut = paprTokenField === Field.OUTPUT;
+
+    if (paprIn) {
       const quoteResult = await getQuoteForSwap(
         ethers.utils.parseUnits(
           amounts.input.toExact(),
           amounts.input.currency.decimals,
         ),
         amounts.input.currency.wrapped.address,
-        amounts.output.currency.wrapped.address,
+        underlying.id,
         tokenName as SupportedToken,
       );
       return quoteResult.sqrtPriceX96After;
-    } else {
+    } else if (paprOut) {
       const quoteResult = await getQuoteForSwapOutput(
         ethers.utils.parseUnits(
           amounts.output.toExact(),
           amounts.output.currency.decimals,
         ),
-        amounts.input.currency.wrapped.address,
+        underlying.id,
         amounts.output.currency.wrapped.address,
         tokenName as SupportedToken,
       );
       return quoteResult.sqrtPriceX96After;
+    } else {
+      return null;
     }
-  }, [amounts, tokenName]);
+  }, [amounts, tokenName, paprTokenField, underlying.id]);
 
   useEffect(() => {
     if (!sqrtPriceAfter || !amounts) return;
     if (!paprTokenField) {
       setPaprPrice(null);
       return;
     }
+    const paprIn = paprTokenField === Field.INPUT;
+    const paprOut = paprTokenField === Field.OUTPUT;
+    const paprUniswapToken = erc20TokenToToken(paprToken, chainId);
+    const underlyingUniswapToken = erc20TokenToToken(underlying, chainId);
+
+    let baseCurrency: Token;
+    let quoteCurrency: Token;
+    if (paprIn && amounts.tradeType === TradeType.EXACT_INPUT) {
+      baseCurrency = paprUniswapToken;
+      quoteCurrency = underlyingUniswapToken;
+    } else if (paprIn && amounts.tradeType === TradeType.EXACT_OUTPUT) {
+      baseCurrency = underlyingUniswapToken;
+      quoteCurrency = paprUniswapToken;
+    } else if (paprOut && amounts.tradeType === TradeType.EXACT_INPUT) {
+      baseCurrency = underlyingUniswapToken;
+      quoteCurrency = paprUniswapToken;
+    } else if (paprOut && amounts.tradeType === TradeType.EXACT_OUTPUT) {
+      baseCurrency = paprUniswapToken;
+      quoteCurrency = underlyingUniswapToken;
+    } else {
+      throw new Error('Invalid trade type');
```

Comment:
what would be this case?

Reply:
added a comment on why this will never happen